### PR TITLE
add support for datetime on final_shifts API parameters

### LINKED ISSUE #3086: Schedules API final shifts endpoint should be filterable by date AND time
### What would you like to see!

The schedules API [final shifts endpoint](https://grafana.com/docs/oncall/latest/oncall-api-reference/schedules/#export-a-schedules-final-shifts) should accept a `datetime` for the `start_date` and `end_date` parameters. This should be consistent with the time ranges available when creating a new schedule.

### Scenario
I have several shift rotations that occur throughout the week (e.g. 9am). I would like to be able to export the shift report for that week **without** including the partial shift that occurs on the first (pre 9am) and last date (post 9am). This is not currently possible since the filters only support rounding to the nearest full day.

### Product Area

Schedules

### Anything else to add?

_No response_

### PR DESCRIPTION
**Pull request by njohnstone2**
# What this PR does
- Changes the data type from `DateField` to `DateTimeField` on the `final_shifts` API endpoint
- Accepts either a date or a datetime for the `start_date` and `end_date` parameters (e.g. 2021-01-01 or 2021-01-01T01:00)
- Datetime granularity is in line with what is configurable on the schedule i.e. `YYYY-MM-DD hh:mm`
- removes the rounding logic that modifies the query sent on the database

## Which issue(s) this PR fixes
#3086 

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

**Review by joeyorlando — COMMENTED** [review]
overall LGTM. Thanks for the contribution! Once you update those tests I'll go ahead and approve/merge

**Comment by joeyorlando** [comment]
@njohnstone2 also do you mind adding a small note about this to the changelog? thank you!

File: engine/apps/public_api/tests/test_schedules.py [hunk 1]
```diff
@@ -1095,7 +1102,7 @@ def test_oncall_shifts_export_truncate_events(
 
     # request shifts on a Tu (ie. 00:00 - 09:00)
     url = reverse("api-public:schedules-final-shifts", kwargs={"pk": schedule.public_primary_key})
-    response = client.get(f"{url}?start_date=2023-01-03&end_date=2023-01-03", format="json", HTTP_AUTHORIZATION=token)
+    response = client.get(f"{url}?start_date=2023-01-03T09:00&end_date=2023-01-03T09:00", format="json", HTTP_AUTHORIZATION=token)
     assert response.status_code == status.HTTP_200_OK
 
     expected_on_call_times = {user1_public_primary_key: 9}
```
[the remark this answers is not in the document]
Comment (njohnstone2) [reply engine/apps/public_api/tests/test_schedules.py:1126]: I added separate tests for validating the parameter formats. Let me know if i missed the mark on what you were asking. 

This change essentially means that specifying a date only will result in a report that starts and ends at 00:00 for the time.

**Comment by njohnstone2** [comment]
@joeyorlando Is there any update on this PR? Just making sure you are not waiting on anything from me

**Comment by joeyorlando** [comment]
@njohnstone2 apologies, I was out for a bit. Merging this now. Thanks again for the contribution 🙌

**Review by joeyorlando — APPROVED** [review]

**Comment by njohnstone2** [comment]
@joeyorlando no worries at all. Have you got any ideas why the `Grafana: 10.0.2 (pull_request)` job failed? I'm. not quite sure how to debug it as there are no logs.

**Pull request merged**